common/src/utils/lifecycle.rs

```diff
@@ -0,0 +1,32 @@
+use dioxus::prelude::*;
+
+pub struct LifeCycle<D: FnOnce()> {
+    ondestroy: Option<D>,
+}
+
+/// It works like a useEffect hook, but it will be called only once
+/// when the component is mounted
+/// and when the component is unmounted
+pub fn use_component_lifecycle<C: FnOnce() + 'static, D: FnOnce() + 'static>(
+    cx: &ScopeState,
+    create: C,
+    destroy: D,
+) -> &LifeCycle<D> {
+    cx.use_hook(|| {
+        cx.spawn(async move {
+            // This will be run once the component is mounted
+            std::future::ready::<()>(()).await;
+            create();
```

Comment:
Something ive overlooked here is that youre calling this in a async task. While im not sure what the context behind it might be for every use case, if blocking functions are called, it might be best to spawn a blocking task here so it does not block the executor